feature-odyssey/src/main/kotlin/com/hedvig/android/odyssey/input/ui/audiorecorder/AudioRecorder.kt

```diff
@@ -0,0 +1,257 @@
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.CircularProgressIndicator
+import androidx.compose.material.IconButton
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.hedvig.android.core.designsystem.component.button.LargeContainedTextButton
+import com.hedvig.android.core.designsystem.component.button.LargeTextButton
+import com.hedvig.android.odyssey.R
+import com.hedvig.android.odyssey.repository.AutomationClaimInputDTO2
+import com.hedvig.odyssey.remote.file.File
+import com.hedvig.odyssey.remote.file.FileContent
+import com.hedvig.odyssey.renderers.audiorecorder.PlaybackWaveForm
+import com.hedvig.odyssey.renderers.audiorecorder.RecordingAmplitudeIndicator
+import com.hedvig.odyssey.renderers.utils.ScreenOnFlag
+import com.hedvig.odyssey.utils.contentType
+import java.time.Clock
+import java.time.Duration
+import java.time.Instant
+import kotlinx.coroutines.launch
+import org.koin.androidx.compose.getViewModel
+
+@Composable
+fun AudioRecorderScreen(
+  questions: List<AutomationClaimInputDTO2.AudioRecording.AudioRecordingQuestion>,
+  onAudioFile: suspend (File) -> Unit,
+  onNext: () -> Unit,
+) {
+  val audioRecorderViewModel = getViewModel<AudioRecorderViewModel>()
+  val audioRecorderViewState by audioRecorderViewModel.viewState.collectAsState()
+
+  val coroutineScope = rememberCoroutineScope()
+
+  Box(
+    Modifier
+      .fillMaxHeight()
+      .padding(all = 16.dp),
+  ) {
+
+    Column {
+      questions.forEach {
+        Surface(
+          elevation = 2.dp,
+          modifier = Modifier.padding(vertical = 4.dp),
+          shape = RoundedCornerShape(20),
+        ) {
+          Text(
+            text = it.getText(),
+            modifier = Modifier.padding(12.dp),
+            fontSize = 16.sp,
+          )
+        }
+      }
+    }
+
+    Box(Modifier.align(Alignment.BottomCenter)) {
+      AudioRecorder(
+        viewState = audioRecorderViewState,
+        startRecording = audioRecorderViewModel::startRecording,
+        clock = audioRecorderViewModel.clock,
+        stopRecording = audioRecorderViewModel::stopRecording,
+        submit = {
+          val filePath = (audioRecorderViewState as? AudioRecorderViewModel.ViewState.Playback)!!.filePath
+          val audioFile = File(
+            name = "AudioRecording",
+            content = FileContent(path = filePath),
+            contentType = filePath.contentType(),
+          )
```

Comment:
This also makes me think this could live somewhere outside of the composable, creating the File object and so on.
This is of course one of these cases of us copying it over from how it was done in Odyssey, but just noting these down so that we consider changing it later since here we don't have the constraints that we had there.